src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/convert/MappingDocumentDbConverter.java

```diff
@@ -47,6 +48,47 @@ public MappingDocumentDbConverter(
         this.objectMapper = objectMapper == null ? ObjectMapperFactory.getObjectMapper() : objectMapper;
     }
 
+    /**
+     * Convert a property value to the value stored in CosmosDB
+     *
+     * @param fromPropertyValue
+     * @return
+     */
+    public static Object toDocumentDBValue(Object fromPropertyValue) {
+        if (fromPropertyValue == null) {
+            return null;
+        }
+
+        // com.microsoft.azure.documentdb.JsonSerializable#set(String, T) cannot set values for Date and Enum correctly
+        if (fromPropertyValue instanceof Date) {
+            fromPropertyValue = ((Date) fromPropertyValue).getTime();
+        } else if (fromPropertyValue instanceof Enum) {
+            fromPropertyValue = fromPropertyValue.toString();
+        }
+
+        return fromPropertyValue;
+    }
+
+    public <T extends Object> T readAsync(Class<T> domainClass,
+                                          @NonNull com.microsoft.azure.cosmosdb.Document document) {
+        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```

Comment:
Configure this earlier if the  property is required? As this configuration already changes the state for the objectMapper  field.

Reply:
Yes, will move to constructor.